tslib/src/functions-throttled/sync-results-to-sql/sync-results.data.service.ts

```diff
@@ -36,6 +42,104 @@ export class SyncResultsDataService implements ISyncResultsDataService {
+    const checkParams = new Array<ISqlParameter>()
+    checkParams.push({
+      name: 'checkCode',
+      value: markedCheck.checkCode,
+      type: TYPES.UniqueIdentifier
+    })
+    const updateCheckRecord: ITransactionRequest = {
+      params: checkParams,
+      sql: checkSql
+    }
+
+    const checkIdQueryResult = await this.sqlService.query('SELECT id FROM [mtc_admin].[check] WHERE checkCode=@checkCode',
+      [{ name: 'checkCode', value: markedCheck.checkCode, type: TYPES.UniqueIdentifier }])
+    const checkId = checkIdQueryResult[0].id
+
+    const errorLogSql = 'INSERT INTO [mtc_results].[checkResultSyncError] (check_id, errorMessage) VALUES (@checkId, @errorMessage)'
+    const errorLogParams = new Array<ISqlParameter>()
+    errorLogParams.push({
+      name: 'checkId',
+      type: TYPES.Int,
+      value: checkId
+    })
+    errorLogParams.push({
+      name: 'errorMessage',
+      type: TYPES.NVarChar(),
+      value: errorMessage
+    })
+    const insertErrorLog: ITransactionRequest = {
+      params: errorLogParams,
+      sql: errorLogSql
+    }
+    return this.sqlService.modifyWithTransaction([updateCheckRecord, insertErrorLog])
+  }
+
+  /**
+   * Delete the existing check result row and associated records from the mtc_results schema
+   * @param {MarkedCheck} markedCheck
+   */
+  public async deleteExistingResultIfExists (markedCheck: MarkedCheck): Promise<void> {
+    const checkInfoParams = new Array<ISqlParameter>()
+    checkInfoParams.push({
+      name: 'checkCode',
+      value: markedCheck.checkCode,
+      type: TYPES.UniqueIdentifier
+    })
+    const checkResultInfoQueryResult = await this.sqlService.query(`
+      SELECT cr.id as [checkResultId], cr.userDevice_id as [userDeviceId]
+      FROM [mtc_results].[checkResult] cr
+      INNER JOIN [mtc_admin].[check] chk ON cr.check_id = chk.id
+      WHERE chk.checkCode = @checkCode`, checkInfoParams)
+    const checkResultInfo = checkResultInfoQueryResult[0]
+    if (checkResultInfo === undefined) return
+    const sql = `
+      BEGIN TRANSACTION
+        DELETE FROM [mtc_results].[userInput] WHERE answer_id IN (
+          SELECT id FROM [mtc_results].[answer] WHERE checkResult_id = @checkResultId
+        )
+        DELETE FROM [mtc_results].[answer] WHERE checkResult_id = @checkResultId
+        DELETE FROM [mtc_results].[event] WHERE checkResult_id = @checkResultId
+        DELETE FROM [mtc_results].[checkResult] WHERE id = @checkResultId
+        DELETE FROM [mtc_results].[userDevice] WHERE id = @userDeviceId
+      COMMIT TRANSACTION
+    `
+    const params = new Array<ISqlParameter>()
+    params.push({
+      name: 'checkResultId',
+      type: TYPES.Int,
+      value: checkResultInfo.checkResultId
+    })
+    params.push({
+      name: 'userDeviceId',
+      type: TYPES.Int,
+      value: checkResultInfo.userDeviceId
+    })
+    return this.sqlService.modify(sql, params)
```

Comment:
I would have expected `modifyWIthTransaction()` but the code above does not have a TRY handler or explicit Rollback. Curious as to why?

Reply:
good idea, keeps it consistent with the rest